common/src/index.ts

```diff
@@ -24,18 +24,15 @@ export type PickWhere<T, U> = Pick<
 >;
 
 // Construct object from properties of T that do not extend U.
-export type OmitWhere<T, U> = Omit<
-  T,
-  {
-    [K in keyof T]: T[K] extends U ? K : never;
-  }[keyof T]
->;
+export type OmitWhere<T, U> = {
+  [K in keyof T as Required<T>[K] extends U ? never : K]: T[K];
+};
 
 // If K equals I return never, otherwise return the key.
 export type FilterKey<K, I> = A.Equals<K, I> extends 1 ? never : K;
 
 // Omit index signature of T if it equals index-signature I.
-export type OmitIndex<T, I extends string | number> = {
```

Comment:
This could potentially result in something different after the change if I am not mistaken. Did you test it?